cmd/influx/write_test.go

```diff
@@ -371,128 +371,50 @@ func Test_fluxWriteDryrunF(t *testing.T) {
-	prevHost := flags.host
-	prevToken := flags.token
-	defer func() {
-		flags.host = prevHost
-		flags.token = prevToken
-	}()
-	useTestServer := func() {
-		httpClient = nil
-		lineData = lineData[:0]
-		flags.token = "myToken"
-		flags.host = server.URL
-	}
-
+func Test_writeRunE(t *testing.T) {
 	t.Run("validates that --org or --org-id must be specified", func(t *testing.T) {
-		t.Skip(`this test is hard coded to global variables and one small tweak causes a lot of downstream test failures changes else, skipping for now`)
-		useTestServer()
 		command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
 		command.SetArgs([]string{"--format", "csv"})
 		err := command.Execute()
 		require.Contains(t, fmt.Sprintf("%s", err), "org")
 	})
 
 	t.Run("validates that either --bucket or --bucket-id must be specified", func(t *testing.T) {
-		useTestServer()
 		command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
 		command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket", "--bucket-id", "my-bucket"})
 		err := command.Execute()
 		require.Contains(t, fmt.Sprintf("%s", err), "bucket") // bucket or bucket-id, but not both
 	})
 
 	t.Run("validates --precision", func(t *testing.T) {
-		useTestServer()
 		command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
 		command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket", "--precision", "pikosec"})
 		err := command.Execute()
 		require.Contains(t, fmt.Sprintf("%s", err), "precision") // invalid precision
 	})
 
-	t.Run("validates --host must be supplied", func(t *testing.T) {
-		t.Skip(`this test is hard coded to global variables and one small tweak causes a lot of downstream test failures changes else, skipping for now`)
-		useTestServer()
-		flags.host = ""
-		command := cmdWrite(&flags, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
-		command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket"})
-		err := command.Execute()
-		require.Contains(t, fmt.Sprintf("%s", err), "host")
-	})
-
 	t.Run("validates decoding of bucket-id", func(t *testing.T) {
-		useTestServer()
 		command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
 		command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket-id", "my-bucket"})
 		err := command.Execute()
 		require.Contains(t, fmt.Sprintf("%s", err), "bucket-id")
 	})
 
 	t.Run("validates decoding of org-id", func(t *testing.T) {
-		useTestServer()
 		command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard, viper: viper.New()})
 		command.SetArgs([]string{"--format", "csv", "--org-id", "my-org", "--bucket", "my-bucket"})
 		err := command.Execute()
 		require.Contains(t, fmt.Sprintf("%s", err), "org-id")
 	})
 
-	t.Run("validates error when failed to retrieve buckets", func(t *testing.T) {
```

Comment:
These 3 are no longer relevant (at least for unit tests). The write API accepts both IDs and names in write requests (I assume because resolving an ID from a name requires read-access on the target resource, which writers might not have). It's up to the server to do the right thing with all the parameters.

I'll add a replacement case for a generic failure-on-write.